### PR TITLE
IGNITE-23196 [ducktests] Change JVM options to JDK11 ones

### PR DESCRIPTION
**Pull request by okreda1**
- changed default settings for gc log (-Xloggc ->-Xlog:gc)
- added new logs for safepoints
- added new variable to disable safepoints log
- removed UnlockCommercialFeatures because it was used only for jfr, but no longer needed for openjdk11
- changed option in check_boolean_options__go_after_default_ones_and_overwrite_them__if_passed_via_jvm_opt to FlightRecorder
- changed option in check_colon_options__go_after_default_ones_and_overwrite_them__if_passed_via_jvm_opt to -Xlog:gc

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/ducktests/tests/checks/utils/check_ignite_spec.py [hunk 1]
```diff
@@ -79,19 +78,10 @@ def check_default_jvm_options__are_not_used__if_merge_with_default_is_false(serv
     assert len(spec.jvm_opts) == 0
 
 
-def check_boolean_options__go_after_default_ones_and_overwrite_them__if_passed_via_jvm_opt(service):
```
Comment (skorotkov) [inline modules/ducktests/tests/checks/utils/check_ignite_spec.py:82]: Even if UnlockCommercialFeatures option is not used for JVM 11+  anymore it's not a reason to drop the test for the boolean options support.  Just choose another boolean option.  The -XX:+FlightRecorder would be ok I think.

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py [hunk 2]
```diff
@@ -102,6 +102,10 @@ def __get_default_jvm_opts(self):
                                                oom_path=os.path.join(self.service.log_dir, "out_of_mem.hprof"),
                                                vm_error_path=os.path.join(self.service.log_dir, "hs_err_pid%p.log"))
 
+        if not self.service.context.globals.get(SAFEPOINT_LOGS_DISABLED, False):
+            default_jvm_opts = merge_jvm_settings(
+                default_jvm_opts, ["-Xlog:safepoint:" + os.path.join(self.service.log_dir, "safepoint.log")])
```
Comment (skorotkov) [inline modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py:107]: Please use more readable settings for log file like:

-Xlog:safepoint*=debug:file=/mnt/service/logs/safepoint.log:time,uptime,level,tags

File: modules/ducktests/tests/ignitetest/services/utils/jvm_utils.py [hunk 3]
```diff
@@ -31,14 +31,14 @@
 
 
 def create_jvm_settings(heap_size=DEFAULT_HEAP, gc_settings=JVM_PARAMS_GC_G1, generic_params=JVM_PARAMS_GENERIC,
-                        gc_dump_path=None, oom_path=None, vm_error_path=None):
+                        gc_dump_path=None, oom_path=None, vm_error_path=None,):
```
Comment (skorotkov) [inline modules/ducktests/tests/ignitetest/services/utils/jvm_utils.py:34]: Why this change?

File: modules/ducktests/tests/ignitetest/utils/ignite_test.py [hunk 4]
```diff
@@ -27,6 +27,7 @@
 # globals:
 JFR_ENABLED = "jfr_enabled"
 IGNITE_TEST_CONTEXT_CLASS_KEY_NAME = "IgniteTestContext"
+SAFEPOINT_LOGS_DISABLED = "SafePointLogsDisabled"
```
Comment (skorotkov) [inline modules/ducktests/tests/ignitetest/utils/ignite_test.py:30]: Please use the snake case for global variable name (in other words follow the existing approach of already existing variables: jfr_enabled, ignite_versions, jmx_remote etc.)

Comment (skorotkov) [inline modules/ducktests/tests/ignitetest/utils/ignite_test.py:30]: Please also change it *_enabled (with True by default). In all other variables we have enabled but not disabled.

Say "safepoint_log_enabled".

File: modules/ducktests/tests/checks/utils/check_ignite_spec.py (hunk 1 above)
Comment (okreda1) [reply modules/ducktests/tests/checks/utils/check_ignite_spec.py:82]: Returned the test, tweaked it to check FlightRecorder in options

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py (hunk 2 above)
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py]: Done

File: modules/ducktests/tests/ignitetest/services/utils/jvm_utils.py [hunk 5]
```diff
@@ -31,14 +31,14 @@
...
     """
     Provides settings string for JVM process.
     param opts: JVM options to merge. Adds new or rewrites default values. Can be list or string.
     """
     gc_dump = ""
     if gc_dump_path:
-        gc_dump = "-verbose:gc -Xloggc:" + gc_dump_path
+        gc_dump = "-Xlog:gc:" + gc_dump_path
```
[the remark this answers is not in the document]
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/services/utils/jvm_utils.py]: Done

File: modules/ducktests/tests/ignitetest/services/utils/jvm_utils.py (hunk 3 above)
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/services/utils/jvm_utils.py]: By mistake, removed it

File: modules/ducktests/tests/ignitetest/utils/ignite_test.py (hunk 4 above)
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/utils/ignite_test.py]: Done

Comment (okreda1) [reply modules/ducktests/tests/ignitetest/utils/ignite_test.py]: Done

File: modules/ducktests/README.md [hunk 6]
```diff
@@ -77,24 +77,91 @@ You may set versions (products) using `@ignite_versions` decorator at code
...
+            "key_store_jks": "admin.jks",
+            "key_store_password": "123456",
+            "trust_store_jks": "truststore.jks",
+            "trust_store_password": "123456"
+          }
+        }
+      }
+    }
+```
+Options `key_store_jks` and `trust_store_jks` is paths.
```
Comment (skorotkov) [inline modules/ducktests/README.md:136]: ...  **are** paths

File: modules/ducktests/README.md [hunk 7]
```diff
@@ -77,24 +77,91 @@ You may set versions (products) using `@ignite_versions` decorator at code
...
+If you start it with `/` it will be used as an absolute path, otherwise as a relative path.
```
Comment (skorotkov) [inline modules/ducktests/README.md:137]: Relative to which directory ?

File: modules/ducktests/tests/checks/utils/check_ignite_spec.py [hunk 8]
```diff
@@ -81,17 +81,19 @@ def check_default_jvm_options__are_not_used__if_merge_with_default_is_false(serv
 
 def check_boolean_options__go_after_default_ones_and_overwrite_them__if_passed_via_jvm_opt(service):
     service.context.globals[JFR_ENABLED] = True
-    spec = IgniteApplicationSpec(service, jvm_opts="-XX:-UnlockCommercialFeatures")
-    assert "-XX:-UnlockCommercialFeatures" in spec.jvm_opts
-    assert "-XX:-UnlockCommercialFeatures" in spec.jvm_opts
-    assert spec.jvm_opts.index("-XX:-UnlockCommercialFeatures") >\
-           spec.jvm_opts.index("-XX:+UnlockCommercialFeatures")
+    spec = IgniteApplicationSpec(service, jvm_opts="-XX:-FlightRecorder")
+    assert "-XX:-FlightRecorder" in spec.jvm_opts
```
Comment (skorotkov) [inline modules/ducktests/tests/checks/utils/check_ignite_spec.py:85]: We have here 2 identical asserts.

I think one of them should be replaced with the 
`assert "-XX:+FlightRecorder" in spec.jvm_opts`

Otherwise it doesn't make sense.

File: modules/ducktests/README.md (hunk 7 above)
Comment (okreda1) [reply modules/ducktests/README.md]: Specified

File: modules/ducktests/README.md (hunk 6 above)
Comment (okreda1) [reply modules/ducktests/README.md]: Fixed

File: modules/ducktests/tests/ignitetest/utils/ignite_test.py (hunk 4 above)
Comment (skorotkov) [reply modules/ducktests/tests/ignitetest/utils/ignite_test.py]: just for record 

After all it was decided to have it False by default to not increase total logs size in results for tests which do not need it

**Review by skorotkov — APPROVED** [review]
